test/src/main/java/org/apache/accumulo/test/compaction/ExternalCompactionProgressIT.java

```diff
@@ -0,0 +1,166 @@
+    try (AccumuloClient client =
+        Accumulo.newClient().from(getCluster().getClientProperties()).build()) {
+      ExternalCompactionTestUtils.createTable(client, table1, "cs1");
+      ExternalCompactionTestUtils.writeData(client, table1, ROWS);
+      c1 = ((MiniAccumuloClusterImpl) getCluster()).exec(Compactor.class, "-q", "DCQ1");
+      coord = ExternalCompactionTestUtils.startCoordinator(((MiniAccumuloClusterImpl) getCluster()),
+          CompactionCoordinator.class, getCluster().getServerContext());
+
+      Thread checkerThread = startChecker();
+      checkerThread.start();
+
+      IteratorSetting setting = new IteratorSetting(50, "Slow", SlowIterator.class);
+      SlowIterator.setSleepTime(setting, 1);
+      client.tableOperations().attachIterator(table1, setting,
+          EnumSet.of(IteratorUtil.IteratorScope.majc));
+      log.info("Compacting table");
+      ExternalCompactionTestUtils.compact(client, table1, 2, "DCQ1", true);
+      ExternalCompactionTestUtils.verify(client, table1, 2, ROWS);
+
+      log.info("Done Compacting table");
+      compactionFinished.set(true);
+      checkerThread.join();
+
+      verifyProgress();
+    } finally {
+      ExternalCompactionTestUtils.stopProcesses(c1, coord);
+    }
+  }
+
+  public Thread startChecker() {
+    return Threads.createThread("RC checker", () -> {
+      try {
+        while (!compactionFinished.get()) {
+          checkRunning();
+          sleepUninterruptibly(1000, TimeUnit.MILLISECONDS);
+        }
+      } catch (TException e) {
+        log.warn("{}", e.getMessage(), e);
+      }
+    });
+  }
+
+  /**
+   * Check running compaction progress.
+   */
+  private void checkRunning() throws TException {
+    var ecList = ExternalCompactionTestUtils.getRunningCompactions(getCluster().getServerContext());
+    var ecMap = ecList.getCompactions();
+    if (ecMap != null) {
+      ecMap.forEach((ecid, ec) -> {
+        // returns null if it's a new mapping
+        var rci = new RunningCompactorInfo(System.currentTimeMillis(), ecid, ec);
+        var previous = runningMap.put(ecid, rci);
+        if (previous == null) {
+          log.debug("New ECID {} with inputFiles: {}", ecid, rci.inputFiles);
+        } else {
+          log.debug("{} progressed from {} to {}", ecid, previous.progress, rci.progress);
+          if (rci.progress <= previous.progress) {
+            log.warn("Compaction did not progress. It went from {} to {}", previous.progress,
+                rci.progress);
+          } else {
+            if (rci.progress > 0 && rci.progress < 25)
+              progressList.add(EC_PROGRESS.STARTED);
+            else if (rci.progress > 25 && rci.progress < 50)
+              progressList.add(EC_PROGRESS.QUARTER);
+            else if (rci.progress > 50 && rci.progress < 75)
+              progressList.add(EC_PROGRESS.HALF);
+            else if (rci.progress > 75 && rci.progress < 100)
+              progressList.add(EC_PROGRESS.THREE_QUARTERS);
```

Comment:
what if progress is 0, 25, 50, 75 or 100?

Reply:
There won't ever be an update for 100% or 0% since these states are represented by other actions instead of passing updates between the processes. I will add comparisons to catch the other values.